README
74 Implement hclust stats and plot buttons.
73 Add a Dendrogram button to hclust.
72 Multiple risk charts now on one plot.
71 RPart does not display tree by default - use the button.
69 Add new options for KMeans. Generate random numbers for seeds.
68 Implement export of Log tab.
65 Move to multiple R src files and modularisation. Implement New button.
64 Initial implementation of paradigms.
63 Tools menu to navigate tabs (Anthony Nolan).
60 RF sampsize implemented (Stuart Hamilton)
59 Add varImpPlot to RF.
55 Support different summary options.
54 Fix plot colours in ROCR plots.
53 Add ggobi option to Explore tab (Marco Lo).
50 All packages loaded with quietly=TRUE.
48 All packages loaded with require.
47 Don't run examples in evaluateRisk.Rd (fails on CRAN MS/Windows).
45 Remove e1071's svm. Allow multiple model evaluation (Stuart Hamilton).
44 Allow target to be a weight variable (Liyin Xue).
43 Add a seed to the sampling for repeatable modelling (Liyin Zue).
42 Make ksvm the default SVM.
40 Add a biplot to principal components.
39 Remove seriation plot---too slow.
38 KMeans GUI now allows specifying k (Weiqiang Lin). 
37 Multiple plots per page.
29 Add a density plot to histograms (John Maindonald).
26 Add line plots for Benford's law (Tatiana Semenova).
23 Add Missing Value correlations (Ray Lindsay).
20 Add RData to file filter in loading Rdata files.
19 The default working directory follows latest openned file.
18 A Sample checkbutton has been added to the Explore tab.
14 Moved to hosting at google code.

NEWS
Refer to the change log for recent news.

Changes in 2.5.20 2010-01-31 11:08:20 Graham Williams
-----------------------------------------------------

The weather dataset disappeared - restore it.


Changes in 2.5.19 2010-01-30 10:33:50 Graham Williams
-----------------------------------------------------

This is mainly minor tweaking and bug fixing.

Turn translations off for MS/Windows until the RGtk2 package is
updated to 2.12.18. Michael Lawrence is adding a binding to
rgtk2_bindtextdomain so that under MW/Windows we can get Glade to also
access the right libraries for internationalisation. Works just fine
under Linux.

The weather dataset is being cleaned up and made more complete, to
capture data from more weather stations in Australia.


Changes in 2.5.18 2010-01-23 12:13:45 Graham Williams
-----------------------------------------------------

Bug fix - left unwanted print statements. Need to pay more attention
to John Chambers!


Changes in 2.5.17 2010-01-22 22:41:05 Graham Williams
-----------------------------------------------------

Some minor fixes.

More work on translations, particularly the user interface elements
through the glade file.


Changes in 2.5.16 2010-01-22 12:11:25 Graham Williams
-----------------------------------------------------

Summary of recent changes, as reported to the rattle-user group at
Google groups.

This version includes: 

A move from sampling into just two datasets to the more traditional 
three datasets: Train/Validate/Test. The Train dataset is used as 
before for building models. The Validate is used to measure the 
performance of the model and then for the user to tune the model 
building parameters, to build the next "better" model, etc. The Test 
dataset is then a true hold out dataset (not used at all in the 
process otherwise) so a better unbiased estimate of error is obtained. 
The user can now specify partitions as 70/15/15, or 50/25/25, etc. Or 
70/30 (or just 70) to get previous behaviour (i.e., only a Train/Test 
dataset). 

Ongoing markup of all text messages within Rattle to support 
translations to other human languages. The framework for this is 
already in place now, and some sample but limited translations are 
available. For example, on GNU/Linux (at least), try: 

> Sys.setenv(LANGUAGE="ja") 
> library(rattle) 
> rattle() 

for some Japanese translations. Offers to translate to other languages 
are most welcome. 

An initial implementation of biclustering under the Cluster tab. 
Biclustering looks for the best clustering of subsets of the data. It 
subsets both the observations and the variables. Only very basic 
support for now, but the interface will grow to provide choice in the 
algorithm used and options associated with each algorithm. The clara 
algorithm for clustering is also expected to be supported soon. See 
http://bm2.genes.nig.ac.jp/RGM2/pkg.php?p=biclust and 
http://www.statistik.uni-dortmund.de/useR-2008/slides/Kaiser+Leisch.pdf 

Experimental support for conditional trees and conditional random 
forests using the party package. Conditional trees and forests are 
newer approaches to building decision trees and address various bias 
issues. See http://journal.r-project.org/archive/2009-2/RJournal_2009-2_Strobl~et... 

Survival modelling. 

Direct read of an Excel spreadsheet on MS/Windows. On the Data tab, 
the CSV option is now labelled Spreadsheet. Identify a .xls or .xlsx 
file, and Rattle will read the data from the spreadsheet. This is 
still experimental. If it gets used I'll improve it. Currently, only 
the first sheet is loaded, and only if it is called Sheet1 (the 
default), and only if the data starts in row 1. 

Tooltips re-enabled for GNU/Linux. Most users will now be using new 
versions of the GTK libraries. The older version was having issues 
with tooltips on GNU/Linux. 

Improved PMML support. Togaware also provides modules to export PMML 
models as standalone C code - this is not open source yet (it will be 
eventually). The functionality is commercially available as part of 
the RStat offering through Information builder's WebFocus product, and 
from Togaware. PMML models can also be depoyed on the Amazon cloud 
through Zementis' offerings. 

Experimental support for ggplot2. This is a little hidden for now, 
but a pointer for future development. 

See the old change log for more history.

src/zzz.R
.onLoad <- function(libname, pkgname)
{
  # 090315 Create the crs environment here. It is defined here (and
  # then also reset in rattle() so that R CMD check would not complain
  # about knowing nothing of crs (after removing the crs<<- assigments
  # throughout rattle)!

  crs <<- new.env()

  # 090207 Create the global crv environment for Rattle. Once again,
  # this is a deviation from Chamber's Prime Directive, but is akin to
  # the use of option.  It is defined here so that it is glabally
  # known and so that plugins can override options. We generally
  # include here the options that can be overridden by a plugin.
  
  crv <<- new.env()
  
  crv$appname <- "Rattle"
  crv$projext <- ".rattle"
  crv$log.intro <- paste("#", sprintf(Rtxt("%s is Copyright (c) 2006-2010 %s."),
                                      "Rattle", "Togaware Pty Ltd"))
  crv$support.msg <- sprintf(Rtxt("Contact %s."), "support@togaware.com")
  crv$library.command <- "library(rattle)"
  crv$version <- VERSION

  # Some global constants

  # 091130 Use UTF-8 as the default encoding for files. This certainly
  # works okay on GNU/Linux. On Vista I see ISO8859-1 as the default
  # and Acken sees CP932 for Japanese.
  
  crv$csv.encoding <- "UTF-8"
  
  crv$show.timestamp <- FALSE
  ## crv$tooltiphack <- FALSE
  crv$close <- "ask"
  # crv$sample.dataset <- "audit"
  crv$sample.dataset <- "weather"
  
  # 090525 Always load tooltips - now use Settings option to enable on
  # GNU/Linux. 090622 But on older installations we still get the
  # Invalid property error so for now on Unix do not support tooltips.
  
  # 090601 Add the crv$load.tooltips option, so it can be turned off
  # on the command line before starting rattle, since older GTK
  # version has issue: Invalid property tooltip-text!
  
  crv$load.tooltips <- TRUE
  
#100110 testing for Rtxt
#  if (.Platform$OS.type == "unix")
#    crv$load.tooltips <- FALSE # Not working in general on Linux
  
  crv$verbose <- TRUE # Add sub titles to plots ...

  crv$max.categories <- 10 # Above which target assumed numeric, not categoric
  crv$max.vars.correlation <- 40 # Correlation slows down too much
  crv$export.to.c.available <- FALSE # No export to C implemented yet
  crv$show.warnings <- TRUE # 090207 Show test/train warning.
  crv$project.extensions <- c("rattle", "rstat") # Extensions for projects  
  crv$ident.min.rows <- 300 # Unique factors/ints > than this are idents
  crv$default.train.percentage <- 70 # The default sample percentage value.
  crv$default.sample <- "70/15/15" # The default train/validate/test split.
  
  # Log constants

  crv$start.log.comment <- "\n\n# "	# Assume paste with sep=""
  crv$end.log.comment   <- "\n\n"	# Assume paste with sep=""
  
  # Model defaults

  crv$cluster.report.max.obs <- 4000
  crv$scatter.max.vars <- 5
  
  crv$rpart.cp.default        <- 0.010
  crv$rpart.minsplit.default  <- 20
  crv$rpart.minbucket.default <- 7
  crv$rpart.maxdepth.default  <- 30

  crv$ada.ntree.default   <- 50

  crv$rf.ntree.default    <- 500
  crv$rf.mtry.default     <- 10
  crv$rf.sampsize.default <- ""

  # Evaluate

  .RATTLE.DATA <<- NULL
  .RATTLE.SCORE.IN <<- NULL
  .RATTLE.SCORE.OUT <<- NULL
  
  # 090309 We set some other global variables for convenience.

  rattleGUI <<- NULL
  Global_rattleGUI <<- NULL
  viewdataGUI <<- NULL
  on_aboutdialog_response <<- NULL

}

# 080417 The R manual for .onLoad says to use .onAttach for startup
# messages.

.onAttach <- function(libname, pkgname)
{
  # This is executed when the package becomes visible to the user.
  
  # 090206 How to not display the welcome message if quietly=TRUE?
  # Otherwise it is annoying, just like fBasics. randomForest seems to
  # do it correctly? Maybe not.... One solution is for the user to
  # define suppressRattleWelcome and set it to TRUE before loading
  # Rattle. Note that we can't do this with
  # crv$suppress.rattle.wanring, defined in zzz.R of rattle, inside
  # .onLoad, since they get loaded and attached before rstat does.

  if (! exists("suppressRattleWelcome")) suppressRattleWelcome <<- FALSE

  # 091221 The Rtxt does not seem to work from the rattle.R file, so
  # do it here again.
  
  COPYRIGHT <- sprintf(Rtxt("Copyright (c) 2006-2010 %s."), "Togaware Pty Ltd")

  if (! suppressRattleWelcome)
  {

    cat(sprintf(paste(Rtxt("Rattle: Graphical interface for data mining using R."),
                      "\n", Rtxt("Version"),
                      " %s ", COPYRIGHT, "\n", sep=""), VERSION))
    if ("rattle" %in% getOption("defaultPackages"))
      rattle()
    else
      cat(Rtxt("Type 'rattle()' to shake, rattle, and roll your data.\n"))
  }
}
